### PR TITLE
feat: fastapi node health status endpoint BLD-1317

### PR DESCRIPTION
**Pull request by giddyphysicist**
## Summary

Adds ability to import a health status endpoint to an Aineko FastAPI app. 

Endpoint located at route `/health` will return a 200 status code if the FastAPI server is active. 

How to implement -- add the following lines to the FastAPI main app file:
```python
from aineko.extras.fastapi import health_router

app.include_router(health_router)
```
where `app` is the user instance of an Aineko FastAPI app.

## Motivation

Previously, there was not a general method for checking the status for an Aineko pipeline FastAPI node. 

This shows that the FastAPI Aineko node and the pipeline are active. 

## Author Checklist

- [x] Changes are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Documentation is updated if necessary.
- [x] Status checks pass.
- [ ] If the version number is changed, it is updated in `pyproject.toml` , `aineko/__init__.py`, `aineko/templates/first_aineko_pipeline/{{cookiecutter.project_slug}}/pyproject.toml`.

## Reviewer Checklist

- [x] Title is accurate and formatted appropriately.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Docs are updated if necessary.
- [x] Code is pythonic and consistent with the rest of the codebase.
- [x] The code is consistent with the [style guide](https://google.github.io/styleguide/pyguide.html).

### CODE REVIEW

File: aineko/extras/fastapi/health.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# SPDX-License-Identifier: Apache-2.0
+"""Health endpoint for FastAPI server."""
+
+from fastapi import APIRouter
+
+router = APIRouter()
+
+
+@router.get("/health", status_code=200)
+async def read_health() -> dict:
```
Comment (mdigi14) [inline aineko/extras/fastapi/health.py:11]: I don't think this needs to be `async`. We're not using `await` so it doesn't add anything.

File: aineko/extras/fastapi/health.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    The health check endpoint is available via a GET request to the `/health`
+    route and returns a 200 response if the server is active.
+
+    For example, if the FastAPI app is running locally on
+    `http://localhost:8000`, the health check endpoint can be accessed via
+    a GET request query to `http://localhost:8000/health`.
+
+    Example usage in FastAPI app:
+        ```python hl_lines="2 5"
```
Comment (mdigi14) [inline aineko/extras/fastapi/health.py:25]: Is this `hl_lines` syntax related to MKdocs? Im curious how I can preview changes like this on PRs

File: aineko/extras/fastapi/health.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Health endpoint for FastAPI server."""
+
+from fastapi import APIRouter
+
+router = APIRouter()
```
Comment (mdigi14) [inline aineko/extras/fastapi/health.py]: We can probably just call this `health_router` and then use 

`from aineko.extras.fastapi.health import health_router `

File: aineko/extras/fastapi/health.py (hunk 2 above)
Comment (Barabazs) [reply aineko/extras/fastapi/health.py:25]: You can run `make view-docs` locally and then navigate to the page documents this file.
In this case it's to highlight line 2 and 5.

File: aineko/extras/fastapi/health.py (hunk 1 above)
Comment (Barabazs) [reply aineko/extras/fastapi/health.py:11]: I think FastAPI recommends to always use `async def` unless you need to blocking operations.
 
https://fastapi.tiangolo.com/async/#path-operation-functions

File: docs/extras/fastapi.md [hunk 4]
```diff
@@ -11,3 +11,15 @@ The FastAPI extra node can be used by adding the following to `pyproject.toml`
...
+
+# Health Check Endpoint
+
+Aineko comes with a light weight helper function as follows:
+
+::: aineko.extras.fastapi.health.read_health
+
+## Authentication Considerations
+
+By default, the Aineko health endpoint does not contain authentication.
```
Comment (Barabazs) [inline docs/extras/fastapi.md]:
```suggestion
By default, the `/health` endpoint does not contain authentication. 
```
or just
```suggestion
By default, the `health` endpoint does not contain authentication. 
```

Imo we should drop the Aineko here.

File: docs/extras/fastapi.md [hunk 5]
```diff
@@ -11,3 +11,15 @@ The FastAPI extra node can be used by adding the following to `pyproject.toml`
...
+
+Aineko comes with a light weight helper function as follows:
+
+::: aineko.extras.fastapi.health.read_health
+
+## Authentication Considerations
+
+By default, the Aineko health endpoint does not contain authentication. 
+
+One method to add authentication is to use security middleware at the app level for FastAPI. This will handle adding authentication to the health endpoint without having to add custom dependencies.
```
Comment (Barabazs) [inline docs/extras/fastapi.md]: Maybe we should link the FastAPI docs here: https://fastapi.tiangolo.com/advanced/security/

File: docs/extras/fastapi.md
Comment (Barabazs) [inline docs/extras/fastapi.md:1]: I think we should include a short paragraph about the reason to use this health endpoint.

File: aineko/extras/fastapi/health.py (hunk 1 above)
Comment (mdigi14) [reply aineko/extras/fastapi/health.py:11]: @Barabazs Thanks! good to know

File: aineko/extras/fastapi/health.py (hunk 2 above)
Comment (giddyphysicist) [reply aineko/extras/fastapi/health.py:25]: oh, nice. i didn't know we had a make command for that. I had been using `poetry run mkdocs serve` locally

File: docs/extras/fastapi.md
Comment (giddyphysicist) [reply docs/extras/fastapi.md:1]: update here: [42dc84c](https://github.com/aineko-dev/aineko/pull/142/commits/42dc84c8995f32e0b2a788e40c2903ee5a29eb10)

File: docs/extras/fastapi.md (hunk 5 above)
Comment (giddyphysicist) [reply docs/extras/fastapi.md]: updated here: [42dc84c](https://github.com/aineko-dev/aineko/pull/142/commits/42dc84c8995f32e0b2a788e40c2903ee5a29eb10)

File: docs/extras/fastapi.md (hunk 4 above)
Comment (giddyphysicist) [reply docs/extras/fastapi.md]: updated here: [42dc84c](https://github.com/aineko-dev/aineko/pull/142/commits/42dc84c8995f32e0b2a788e40c2903ee5a29eb10)

File: aineko/extras/fastapi/health.py (hunk 3 above)
Comment (giddyphysicist) [reply aineko/extras/fastapi/health.py]: updated here: [42dc84c](https://github.com/aineko-dev/aineko/pull/142/commits/42dc84c8995f32e0b2a788e40c2903ee5a29eb10)

**Review by programmablemike — APPROVED** [review]
👍 LGTM. Thanks for submitting it!

**Review by Barabazs — APPROVED** [review]

**Review by mdigi14 — APPROVED** [review]

**Pull request merged**